docs/architecture.md

```diff
@@ -112,76 +112,206 @@ The redux store is composed of two `slices` : `documents` and `queries`:
+- `loaded`: the query had been run and returned results.
+- `failed`: the last query execution failed.
+
+When a query is called for the first time, it is first initialized in the `queries` collection in the store, with the `pending` status.
 
-### How does it works?
+If the query already exists in the store, its status is checked to ensure that it is not already in a `loading` state.
 
-When a `query` is resolved, CozyClient dispatches a `receiveQueryResult`
-action for a simple `get` but `receiveMutationResult` when we mutate 
-something. 
+Then, the query is "loaded", making its status on `loading`.
 
-### Focus on `receiveMutationResult`: 
+Finally, the query is actually run and the results are retrieved from the database. The status is now `loaded` and some additional information are saved in the query store, such as `lastFetch` and `lastUpdate`.
+If any new data is retrieved by the query, all the documents ids retrieved by the query are then stored in the `data` array. 
 
-Our two slices `documents` and `queries` listen actions and do some specific work on
- `isReceivingMutationResult` action.
+Likewise, the `documents` collection is updated as well with the full documents content.
 
-`documents`: If the `_id` of the mutated document is not present, then we add the document. 
-If the `_id` is already there, then we update the content of the document with the fresh data
-(for now the work is done in extractAndMergeDocument() method). 
 
-So if your app is linked to the documents store via getDocumentFromStore() for instance
-your app will have the updated value. 
+#### Auto query update
 
-`queries` has an autoUpdater mechanism that does something we can explain this way:
-- it takes the mutated documents (newly created or updated) 
-- it converts the initial `query` to a "`js predicate`" (thanks to the [sift library](https://github.com/crcn/sift.js/))
-- For each query already in the `slice` it runs this `js predicate` and detects if the query
-is concerned by the mutation
-- If the query is concerned, then it checks if it has to remove / add the id of the `mutated` 
-document
-(for now the work is done mainly in queries.js/updateData())
+Since there is a link between queries and documents through the ids list in each query object, we need to be able to deal with changes occuring in the documents collection, i.e. when documents are added or deleted.
 
-So in our previous example our `todoslist` is concerned by the addition of the new todo, then 
-the `id` is added to `todolist` data, then the component linked will be refreshed with this new 
-document.
+For this, cozy-client has a mechanism called "query updater": each time a query is run, all the queries in the store are re-evaluated directly in the store.
+That means that for each query, we take its definition, convert it into a mango predicate (see [here](#in-memory-queries-evaluation) for details), and run it against the store's document collection.
```

Comment:
I don't think we talk about `mango` earlier. If we do, maybe we should link to it.